network_test.go

```diff
@@ -0,0 +1,35 @@
+package xmpp
+
+import (
+	"testing"
+)
+
+type params struct {
+}
+
+func TestParseAddr(t *testing.T) {
+	tests := []struct {
+		name  string
+		input string
+		want  string
+	}{
+		{name: "ipv4-no-port-1", input: "localhost", want: "localhost:5222"},
+		{name: "ipv4-with-port-1", input: "localhost:5555", want: "localhost:5555"},
+		{name: "ipv4-no-port-2", input: "127.0.0.1", want: "127.0.0.1:5222"},
+		{name: "ipv4-with-port-2", input: "127.0.0.1:5555", want: "127.0.0.1:5555"},
+		{name: "ipv6-no-port-1", input: "1::", want: "[1::]:5222"},
+		{name: "ipv6-no-port-2", input: "[1::]", want: "[1::]:5222"},
```

Comment:
i prefer `::1` :stuck_out_tongue_winking_eye:

Reply:
Ah, yes, I took some examples online but they are not very good. We should adapt them.

Reply:
Examples are now a bit nicer: https://github.com/FluuxIO/go-xmpp/commit/f79a3a219bfae1368ee12e07e28a9f49c349a500